### PR TITLE
{AKS} Support to enable azure monitor profile when the sku name is automatic

### PR DESCRIPTION
**Pull request by xmzhao0822**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks create -g <resource group> -n <resource name> --sku automatic`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -14,6 +14,9 @@ Pending
 * Add option `WindowsAnnual` to `--os-sku` for `az aks nodepool add`.
 * Add option `--enable-force-upgrade`, `--disable-force-upgrade` and `--upgrade-override-until` to `az aks upgrade`.
 
+3.0.0b10
+* Support to enable azure monitor profile when the sku name is automatic.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:18]:
```suggestion
3.0.0b10
++++++++
* Support to enable azure monitor profile when the sku name is automatic.
* Add option `WindowsAnnual` to `--os-sku` for `az aks nodepool add`.
* Add option `--enable-force-upgrade`, `--disable-force-upgrade` and `--upgrade-override-until` to `az aks upgrade`.
```

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -14,6 +14,12 @@ Pending
 * Add option `WindowsAnnual` to `--os-sku` for `az aks nodepool add`.
 * Add option `--enable-force-upgrade`, `--disable-force-upgrade` and `--upgrade-override-until` to `az aks upgrade`.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]:
```suggestion

```

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=92491328&view=results)
- test_aks_automatic_sku

Will you update the test case to check if azure monitor metrics is enabled?

**Comment by xmzhao0822** [comment]
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=92494002&view=results)
> 
> * test_aks_automatic_sku
> 
> Will you update the test case to check if azure monitor metrics is enabled?

Updated the automatic sku scenario test case to check if azure monitor metrics is enabled.

File: src/aks-preview/HISTORY.rst [hunk 3]
```diff
@@ -11,7 +11,13 @@ To release a new version, please select a new version number (usually plus 1 to
...
 Pending
 +++++++
-* Vendor new SDK and bump API version to 2024-03-02-preview.
+
+3.0.0b11
+++++++++
+* Support to enable azure monitor profile when the sku name is automatic.
+
+3.0.0b10
+++++++++
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:20]:
```suggestion
3.0.0b10
++++++++
* Support to enable azure monitor profile when the sku name is automatic.
```

File: src/aks-preview/setup.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 from setuptools import setup, find_packages
 
-VERSION = "3.0.0b9"
+VERSION = "3.0.0b11"
```
Comment (FumingZhang) [inline src/aks-preview/setup.py:12]:
```suggestion
VERSION = "3.0.0b10"
```
b10 is not released yet

File: src/aks-preview/HISTORY.rst [hunk 5]
```diff
@@ -11,7 +11,10 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
-* Vendor new SDK and bump API version to 2024-03-02-preview.
+
+3.0.0b10
+++++++++
+* Support to enable azure monitor profile when the sku name is automatic.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:17]:
```suggestion
* Support to enable azure monitor profile when the sku name is automatic.
* Vendor new SDK and bump API version to 2024-03-02-preview.
```

**Review by FumingZhang — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Thanks

**Pull request merged**